src/com/projectkorra/projectkorra/PKListener.java

```diff
@@ -1585,7 +1585,7 @@ public void onPlayerSlotChange(final PlayerItemHeldEvent event) {
 		final int slot = event.getNewSlot() + 1;
 		GeneralMethods.displayMovePreview(player, slot);
 
-		if (!ConfigManager.defaultConfig.get().getBoolean("Properties.BendingPreview")) {

```

Comment:
You should probably keep this, but instead create a nested if statement for the waterarms logic. In addition to that I'd move [L1586](https://github.com/ProjectKorra/ProjectKorra/pull/1076/files#diff-34d9e9a9bd1ed2b5ed340633a344ba79R1586)  to within the if statement so that the config option for the bending preview actually does something.

Reply:
Hmm, I'm not 100% sure what you mean, but this is what I have now: https://hastebin.com/heyevohiri.cs

Reply:
Move the statement where it displays the MovePreview into the if statement where it checks if it is enabled or not, so that it obeys the config value.

Reply:
Shouldn't I keep it where it is? If BendingPreview is false, it shouldn't displayMovePreview.

Reply:
So... is this ok? https://hastebin.com/ezazohuxuk.cs

Reply:
Uhh I was thinking more like this

```java
if (ConfigManager.defaultConfig.get().getBoolean("Properties.BendingPreview")) {
			if (ConfigManager.defaultConfig.get().getBoolean("Abilities.Water.WaterArms.DisplayBoundMsg")) {
				final WaterArms waterArms = CoreAbility.getAbility(player, WaterArms.class);
				if (waterArms != null) {
					waterArms.displayBoundMsg(event.getNewSlot() + 1);
					return;
				}
			}
			GeneralMethods.displayMovePreview(player, slot);
		} 

````

Reply:
Idk, I feel like BendingPreview should be independent from DisplayBoundMsg. If it's nested wouldn't that mean changing DisplayBoundMsg won't even do anything unless BendingPreview is true?
How about this:
```
	@EventHandler(priority = EventPriority.NORMAL, ignoreCancelled = true)
	public void onPlayerSlotChange(final PlayerItemHeldEvent event) {
		final Player player = event.getPlayer();
		final int slot = event.getNewSlot() + 1;

		if (ConfigManager.defaultConfig.get().getBoolean("Properties.BendingPreview")) {
			GeneralMethods.displayMovePreview(player, slot);
		}
		
		if (ConfigManager.defaultConfig.get().getBoolean("Abilities.Water.WaterArms.DisplayBoundMsg")) {
			final WaterArms waterArms = CoreAbility.getAbility(player, WaterArms.class);
			if (waterArms != null) {
					waterArms.displayBoundMsg(event.getNewSlot() + 1);
					return;
			}
		}
	}
```

Reply:
That also works.

Reply:
Makes sense, people may want one, both, or neither.